app/interactors/schedule/save_datetime_interactor.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class Schedule::SaveDatetimeInteractor
+  include Interactor
+  delegate :params,
+           :edition,
+           :user,
+           :issues,
+           :datetime,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      check_for_issues
+      set_scheduled_publishing_datetime
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+  end
+
+  def check_for_issues
+    unless edition.draft? || edition.submitted_for_review?
+      raise "Can't save a scheduling date unless edition is a draft or has been submitted for 2i"
+    end
+
+    checker = Requirements::ScheduledDatetimeChecker.new(params[:schedule])
+    pre_issues = checker.pre_submit_issues.issues
+
+    issues = Requirements::CheckerIssues.new(pre_issues + action_issues)
+
+    context.fail!(issues: issues) if issues.any?
+
+    context.datetime = checker.parsed_datetime
+  end
+
+  def set_scheduled_publishing_datetime
+    updater = Versioning::RevisionUpdater.new(edition.revision, user)
+    updater.assign(scheduled_publishing_datetime: context.datetime)
+
+    if updater.changed?
+      edition.assign_revision(updater.next_revision, user).save!
+      create_timeline_entry
+    end
+  end
+
+  def create_timeline_entry
+    TimelineEntry.create_for_revision(
+      entry_type: :scheduled_publishing_datetime_set,
+      edition: edition,
+    )
```

Comment:
I don't think this is needed as per Ben Hazell's comment: https://trello.com/c/5vJhi2a8/691-update-schedule-workflow-based-on-user-research#comment-5cbf2e90ae40db85431fe988 

apologies if that isn't very clear on the trello card I notice the link is only in the description but not the AC

Reply:
Now removed